### PR TITLE
[PL-51451] SMP - combine user profiles and ref arch

### PR DESCRIPTION
**Pull request by brian-f-harness**
Thanks for contributing to the Harness Developer Hub! Our code owners will review your submission.

## Description

* Please describe your changes: SMP - combine user profiles and ref arch
* Jira/GitHub Issue numbers (if any): PL-51451
* Preview links/images (Internal contributors only): [Reference architectures](https://664f9652ed867459a9c3cff9--harness-developer.netlify.app/docs/self-managed-enterprise-edition/reference-architecture)

## PR lifecycle

We aim to merge PRs within one week or less, but delays happen sometimes.

If your PR is open longer than two weeks without any human activity, please tag a [code owner](https://github.com/harness/developer-hub/blob/main/.github/CODEOWNERS) in a comment.

PRs must meet these requirements to be merged:

- [x] Successful preview build.
- [x] Code owner review.
- [x] No merge conflicts.
- [x] Release notes/new features docs: Feature/version released to at least one prod environment.

### CODE REVIEW

**Comment by tuffacton** [comment]
Made a few recs, @brian-f-harness lmk if they make sense or feel free to debate me, then I can review and close

File: src/components/Docs/data/selfManagedEnterpriseEditionData.ts [hunk 1]
```diff
@@ -22,6 +22,13 @@ import {
               "Platforms and technologies supported by the on-prem Harness Self-Managed Enterprise Edition.",
             link: "/docs/self-managed-enterprise-edition/smp-supported-platforms",
           },
+          {
+            title: "Reference Architectures",
```
Comment (brian-f-harness) [inline src/components/Docs/data/selfManagedEnterpriseEditionData.ts:26]:
```suggestion
            title: "Reference architectures",
```

File: src/components/Docs/data/selfManagedEnterpriseEditionData.ts [hunk 2]
```diff
@@ -22,6 +22,13 @@ import {
...
+            module: MODULES.platform,
+            description:
+              "Review Harness Self-Managed Enterprise Edition user profiles and reference architectures.",
```
Comment (brian-f-harness) [inline src/components/Docs/data/selfManagedEnterpriseEditionData.ts:29]:
```suggestion
              "Review Harness Self-Managed Enterprise Edition reference architectures.",
```

**Comment by brian-f-harness** [comment]
> Made a few recs, @brian-f-harness lmk if they make sense or feel free to debate me, then I can review and close



> Made a few recs, @brian-f-harness lmk if they make sense or feel free to debate me, then I can review and close

Makes sense to me @tuffacton, thank you. If you want to approve the PR, I'll merge it after the build succeeds.

**Review by tuffacton — APPROVED** [review]
LGTM

File: docs/self-managed-enterprise-edition/reference-architecture.md [hunk 3]
```diff
@@ -2,32 +2,110 @@
...
+  - /docs/self-managed-enterprise-edition/advanced-configurations/reference-architecture
 ---
 
 import Tabs from '@theme/Tabs';
 import TabItem from '@theme/TabItem';
 
-Harness Self-Managed Enterprise Edition brings a robust and flexible software delivery platform to organizations seeking control over their deployment infrastructure. This reference architecture serves as a guide, outlining the key components and best practices for implementing Harness Self-Managed Enterprise Edition. Organizations can optimize their software delivery platform with this architecture, ensuring reliability, scalability, and consistent software deployments.
+Harness Self-Managed Enterprise Edition brings a robust and flexible software delivery platform to organizations seeking control over their deployment infrastructure. This topic describes user profiles and reference architectures, outlining the key components and best practices for implementing Harness Self-Managed Enterprise Edition. Organizations can optimize their software delivery platform with these and architectures, ensuring reliability, scalability, and consistent software deployments.
```
Comment (douglas-j-bothwell) [inline docs/self-managed-enterprise-edition/reference-architecture.md:14]:
```suggestion
Harness Self-Managed Enterprise Edition brings a robust and flexible software delivery platform to organizations seeking control over their deployment infrastructure. This topic describes user profiles and reference architectures, outlining the key components and best practices for implementing Harness Self-Managed Enterprise Edition. Organizations can optimize their software delivery platform with these architectures, ensuring reliability, scalability, and consistent software deployments.
```

File: docs/self-managed-enterprise-edition/reference-architecture.md [hunk 4]
```diff
@@ -2,32 +2,110 @@
...
+| :-- | :-- | :-- | :--
+| Demo|Up to 10|2|2
+| Small|Up to 200|50|50
+| Medium|Up to 1000|250|250
+| Large|Up to 3000|500|500
 
-Recognizing that each organization has unique requirements, this reference architecture explores customization options and extensibility features. It empowers users to tailor the Harness platform to specific workflows and integrations, ensuring a personalized and efficient continuous delivery pipeline.
+### Demo user requirements
+
+Core CPU and memory requirements depend on the modules you use for demo purposes. For CI and CD, a minimum 3 core CPU and 14 Gi memory are required.
```
Comment (douglas-j-bothwell) [inline docs/self-managed-enterprise-edition/reference-architecture.md:36]:
```suggestion
Core CPU and memory requirements depend on the modules you use for demo purposes. CI and CD require a minimum of 3 core CPUs and 14-Gi of memory.
```

**Review by douglas-j-bothwell — APPROVED** [review]
A few sugs otherwise LGTM

**Pull request merged**